### PR TITLE
Add QtrSensors for line tracking (Pololu QTR Reflectance)

### PR DESCRIPTION
**Pull request by Ellerbach**
## Description

Add QtrSensors for line tracking

## Motivation and Context

- This branch exist for long time but never been tested, a contributor tested it with success

## How Has This Been Tested?

A contributor tested it.

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [x] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/QtrSensors/QtrSensors.nfproj [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectTypeGuids>{11A8DD76-328B-46DF-9F39-F559912D0360};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <ProjectGuid>c13bf8f1-6f45-4925-a3b3-dca3091fb173</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <FileAlignment>512</FileAlignment>
+    <RootNamespace>Iot.Device.QtrSensors</RootNamespace>
+    <AssemblyName>Iot.Device.QtrSensors</AssemblyName>
+    <TargetFrameworkVersion>v1.0</TargetFrameworkVersion>
+    <DocumentationFile>bin\$(Configuration)\Iot.Device.At24cxx.xml</DocumentationFile>
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nfproj:19]: I think this should be:

```suggestion
    <DocumentationFile>bin\$(Configuration)\Iot.Device.QtrSensors.xml</DocumentationFile>
```

File: devices/QtrSensors/QtrSensors.nuspec [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>nanoFramework.Iot.Device.QtrSensors</id>
+    <version>$version$</version>
+    <title>nanoFramework.Iot.Device.QtrSensors</title>
+    <authors>nanoFramework project contributors</authors>
+    <owners>nanoFramework project contributors,dotnetfoundation</owners>
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:8]: This one is deprecated.
```suggestion
```

File: devices/QtrSensors/QtrSensors.nuspec [hunk 3]
```diff
@@ -0,0 +1,49 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>nanoFramework.Iot.Device.QtrSensors</id>
+    <version>$version$</version>
+    <title>nanoFramework.Iot.Device.QtrSensors</title>
+    <authors>nanoFramework project contributors</authors>
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:7]: This has to be:
```suggestion
    <authors>nanoframework</authors>
```

File: devices/QtrSensors/QtrSensors.nuspec [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    <owners>nanoFramework project contributors,dotnetfoundation</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <license type="file">LICENSE.md</license>
+    <releaseNotes>
+    </releaseNotes>
+    <developmentDependency>false</developmentDependency>
+    <projectUrl>https://github.com/nanoframework/nanoFramework.IoT.Device</projectUrl>
+    <iconUrl>https://secure.gravatar.com/avatar/97d0e092247f0716db6d4b47b7d1d1ad</iconUrl>
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:15]: Deprecated
```suggestion
```

File: devices/QtrSensors/QtrSensors.nuspec [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  <metadata>
+    <id>nanoFramework.Iot.Device.QtrSensors</id>
+    <version>$version$</version>
+    <title>nanoFramework.Iot.Device.QtrSensors</title>
+    <authors>nanoFramework project contributors</authors>
+    <owners>nanoFramework project contributors,dotnetfoundation</owners>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <license type="file">LICENSE.md</license>
+    <releaseNotes>
+    </releaseNotes>
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:12]: Missing readme

```suggestion
    </releaseNotes>
    <readme>docs\README.md</readme>
```

File: devices/QtrSensors/QtrSensors.nuspec [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      <dependency id="nanoFramework.System.Device.Adc" version="1.1.6" />
+    </dependencies>
+  </metadata>
+  <files>
+    <file src="bin\Release\Iot.Device.QtrSensors.dll" target="lib\Iot.Device.QtrSensors.dll" />
+    <file src="bin\Release\Iot.Device.QtrSensors.pdb" target="lib\Iot.Device.QtrSensors.pdb" />
+    <file src="bin\Release\Iot.Device.QtrSensors.pdbx" target="lib\Iot.Device.QtrSensors.pdbx" />
+    <file src="bin\Release\Iot.Device.QtrSensors.pe" target="lib\Iot.Device.QtrSensors.pe" />
+    <file src="bin\Release\Iot.Device.QtrSensors.xml" target="lib\Iot.Device.QtrSensors.xml" />
+    <file src="..\Card\bin\Release\Iot.Device.CardRfid.dll" target="lib\Iot.Device.CardRfid.dll" />
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:35]: These Cardif and Mifarre files seem to be wrong here...

File: devices/QtrSensors/QtrSensors.nuspec [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    <file src="..\Card\bin\Release\Iot.Device.CardRfid.pdb" target="lib\Iot.Device.CardRfid.pdb" />
+    <file src="..\Card\bin\Release\Iot.Device.CardRfid.pdbx" target="lib\Iot.Device.CardRfid.pdbx" />
+    <file src="..\Card\bin\Release\Iot.Device.CardRfid.pe" target="lib\Iot.Device.CardRfid.pe" />
+    <file src="..\Card\bin\Release\Iot.Device.CardRfid.xml" target="lib\Iot.Device.CardRfid.xml" />
+    <file src="..\Card\Mifare\bin\Release\Iot.Device.Mifare.dll" target="lib\Iot.Device.Mifare.dll" />
+    <file src="..\Card\Mifare\bin\Release\Iot.Device.Mifare.pdb" target="lib\Iot.Device.Mifare.pdb" />
+    <file src="..\Card\Mifare\bin\Release\Iot.Device.Mifare.pdbx" target="lib\Iot.Device.Mifare.pdbx" />
+    <file src="..\Card\Mifare\bin\Release\Iot.Device.Mifare.pe" target="lib\Iot.Device.Mifare.pe" />
+    <file src="..\Card\Mifare\bin\Release\Iot.Device.Mifare.xml" target="lib\Iot.Device.Mifare.xml" />
+    <file src="readme.md" target="" />
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.nuspec:45]: Missing readme 

```suggestion
    <file src="readme.md" target="" />
    <file src="README.md" target="docs\" />
```

File: devices/QtrSensors/QtrSensors.sln [hunk 8]
```diff
@@ -0,0 +1,35 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.31729.503
```
Comment (josesimoes) [inline devices/QtrSensors/QtrSensors.sln:4]: Better update the SLN file to VS2022 please

File: devices/QtrSensors/samples/Program.cs [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    13 => 14,
+    12 => 15,
+    //14 => 16,
+    14 => 0,
+    27 => 17,
+    25 => 18,
+    //26 => 19,
+    26 => 1,
+    _ => -1
+};
```
Comment (josesimoes) [inline devices/QtrSensors/samples/Program.cs:106]: empty line missing

```suggestion
};

```

**Review by josesimoes — COMMENTED** [review]
Great work adding this!! 👍🏻 
Only minor issues on the projects and build stuff. 😉

File: devices/QtrSensors/QtrSensors.nuspec (hunk 6 above)
Comment (Ellerbach) [reply devices/QtrSensors/QtrSensors.nuspec:34]: ho, bad copy/paste! great catch :-)

**Review by josesimoes — APPROVED** [review]
Perfect! Thanks.

**Pull request merged**